src/routes/controllers/articles.controller.js

```diff
@@ -396,3 +397,60 @@ export async function reportArticle(req, res) {
     );
   }
 }
+
+/**
+ *
+ *
+ * @export
+ * @param {object} req
+ * @param {object} res
+ * @returns {object} rateArticle success/error message and article data
+ */
+export const rateArticle = async (req, res) => {
+  const { rating } = req.body;
+  const { articleId } = req.params;
+  const { id: userId } = req.user;
+
+  try {
+    const article = await Article.findByPk(articleId);
+    if (!article) {
+      return errorResponse(res, 404, 'This article was not found');
+    }
+
+    const [createdRating, isNewRecord] = await Rating.findOrCreate({
+      where: { articleId, userId },
+      defaults: { rating },
+    });
+
+    if (!isNewRecord && createdRating.rating !== rating) {
+      await createdRating.update({
+        rating,
+      });
+    }
+
+    const ratedArticle = await Article.findByPk(articleId, {
+      include: [
+        {
+          model: Rating,
+          as: 'ratings',
+          attributes: ['rating', 'userId'],
+        },
+      ],
+    });
+
+    const jsonRatedArticle = ratedArticle.toJSON();
+    jsonRatedArticle.averageRating = calcAverageRating(
+      jsonRatedArticle.ratings,
+    );
+    delete jsonRatedArticle.ratings;
+
+    return successResponse(
+      res,
+      200,
+      'Your rating has been recorded',
+      jsonRatedArticle,
```

Comment:
The name choice here `jsonratedarticle` sounds a bit off. Can we just call it `article`?

Reply:
I have renamed it to ratedArticleJSON